src/PostSharp.Engineering.BuildTools/Build/Publishers/InvalidatingS3Publisher.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) SharpCrafters s.r.o. See the LICENSE.md file in the root directory of this repository root for details.
+
+using JetBrains.Annotations;
+using PostSharp.Engineering.BuildTools.S3.Publishers;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Net.Http;
+
+namespace PostSharp.Engineering.BuildTools.Build.Publishers;
+
+[PublicAPI]
+public class InvalidatingS3Publisher : S3Publisher
+{
+    public InvalidatingS3Publisher(
+        IReadOnlyCollection<S3PublisherConfiguration> configurations,
+        string invalidationApiKeyEnvironmentVariableName,
```

Comment:
Too convoluted. Accept a URL and use Environment.ExpandEnvironmentVariables.

Reply:
An alternative design is to have a URL as a ParametricString and to extend ParametricString to support environment variables, however it could be more complex to pass the BuildInfo context.